test/e2e/cluster_extension_install_test.go

```diff
@@ -337,6 +269,84 @@ func TestClusterExtensionInstallSuccessorVersion(t *testing.T) {
 	}, pollDuration, pollInterval)
 }
 
+func TestClusterExtensionInstallReResolvesWhenNewCatalog(t *testing.T) {
```

Comment:
@joelanford I wonder if we should rethink the e2es to flow from each other some how and where possible. What I mean is, do we want (just about) every test to start with install a CE then wait? I'm wondering if instead we could 1) install something 2) upgrade that something 3) swap the catalog 4) bring it down, etc. I want to avoid as much as possible the 2h e2e test runtime (if possible). I know there's some pros and cons here. So, just wondering if there's another strategy that we can adopt that keep the resource usage lower while still giving us a strong signal.